### PR TITLE
update interface openWithIde

### PR DESCRIPTION
**Pull request by naturalch**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [x] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: packages/native-pack-tool/src/platforms/windows.ts [hunk 1]
```diff
@@ -64,9 +64,9 @@ export class WindowsPackTool extends NativePackTool {
         return true;
     }
 
-    async openWithIde() {
-        const nativePrjDir = this.paths.nativePrjDir;
-        await toolHelper.runCmake(['--open', `"${cchelper.fixPath(nativePrjDir)}"`]);
+    async openWithIde(options: IOpenWithIdeOptions) {
+        const nativePrjDir = ps.join(options.buildDir, 'proj');
+        await toolHelper.runCmake(['--open', `"${cchelper.fixPath(nativePrjDir)}"`], options.nativeIdeDir);
```
Comment (qiuguohua) [inline packages/native-pack-tool/src/platforms/windows.ts]: Does ‘options.nativeIdeDir‘ mean the directory of the IDE or cmake?

Comment (naturalch) [reply packages/native-pack-tool/src/platforms/windows.ts]: ‘options.nativeIdeDir’ means the directory of the IDE. 'openWithIde' doesn't need to initialize so many unused params, so use 'options' to to specify the required params.

**Review by qiuguohua — APPROVED** [review]

File: packages/native-pack-tool/src/utils.ts [hunk 2]
```diff
@@ -459,8 +459,8 @@ export const toolHelper = {
         });
     },
 
-    runCmake(args: string[]) {
-        let cmakePath = Paths.cmakePath;
+    runCmake(args: string[], customCmakePath?: string) {
+        let cmakePath = customCmakePath || Paths.cmakePath;
```
Comment (yanOO1497) [inline packages/native-pack-tool/src/utils.ts]: why add 'customCmakePath' ?

**Review by bofeng-song — APPROVED** [review]

File: packages/native-pack-tool/src/utils.ts [hunk 3]
```diff
@@ -460,7 +460,7 @@ export const toolHelper = {
     },
 
     runCmake(args: string[]) {
-        let cmakePath = Paths.cmakePath;
+        let cmakePath =  Paths.cmakePath;
```
Comment (qiuguohua) [inline packages/native-pack-tool/src/utils.ts]: extra space

**Review by qiuguohua — APPROVED** [review]

**Review by yanOO1497 — APPROVED** [review]

File: packages/native-pack-tool/src/utils.ts (hunk 3 above)
Comment (naturalch) [inline packages/native-pack-tool/src/utils.ts]:
```suggestion
        let cmakePath = Paths.cmakePath;
```

**Review by VisualSJ — APPROVED** [review]

**Pull request merged**